### DISCUSSION TITLE
How to combine defer/Await with useNavigation loading?

### DISCUSSION THREAD

**Discussion by jrr** [remix-run/remix#9627] [Q&A]
I have a component on my page backed by a slow query, so I throw up a spinner with [defer](https://remix.run/docs/en/main/utils/defer) and [Await](https://remix.run/docs/en/main/components/await), like in the [Streaming guide](https://remix.run/docs/en/main/guides/streaming). This works great for the initial page load.

But there are also some query params that affect the query (e.g. a chart with a date range). These params are altered with `setSearchParams()` from [useSearchParams](https://remix.run/docs/en/main/hooks/use-search-params#usesearchparams).

I'm observing that when the search params are altered, the suspense loading state doesn't appear.

Some related issues suggest using [useNavigation](https://remix.run/docs/en/main/hooks/use-navigation)'s state, but I haven't found an elegant way to combine them.

Is there a way to get a promise out of `useNavigation()`, to trigger the Suspense with? Something like this:
```tsx
<Suspense fallback={...          } >
          <Await resolve={loaderData}>{(loadedData) => { ... }}</Await>
          <Await resolve={useNavigationLoadingPromise}><></></Await>
</Suspense>
```

Does anyone have an example of how they've combined useNavigation state with defer/Await? I want spinners both on the initial load and on subsequent navigation.

https://github.com/remix-run/react-router/issues/10638
https://github.com/remix-run/react-router/issues/10816

**Comment by andrewbarnesweb** [comment]
This behaviour is by design. I got around it (since I'm using SPA version) of the hack of just returning the promise.

https://github.com/remix-run/remix/issues/9572 suggestion is to opt in to the single fetch functionality. (https://github.com/remix-run/remix/issues/9572#issuecomment-2152580203)
